tests/DeepCopyTest/Reflection/ReflectionHelperTest.php

```diff
@@ -29,6 +29,40 @@ public function testMaintainPropertiesKey()
 
         $this->assertSame($expectedProps, array_keys($actualProps));
     }
+
+    /**
+     * @dataProvider propertyDataProvider
+     *
+     * @param string $name
+     */
+    public function testGetProperty($name)
+    {
+        $object = new ReflectionHelperTestChild();
+        $property = ReflectionHelper::getProperty($object, $name);
+        $this->assertInstanceOf('\ReflectionProperty', $property);
+        $this->assertSame($name, $property->getName());
+    }
+
+    /**
+     * @return array
+     */
+    public function propertyDataProvider()
+    {
+        return [
+            'public property' => ['a4'],
+            'private property' => ['a9'],
+            'private property of ancestor' => ['a3']
```

Comment:
@theofidry i am not @trebi but I can give you one use-case:
If you work with entities handled by doctrine it is nearly impossible to get all the time the pure entity back. So the ProxyEntity is a Child of the actual entity. But the 'real' properties from the actual entity are most of the time private. 
If we would implement what is proposed in [issue#67](https://github.com/myclabs/DeepCopy/issues/67) we could forward the 'power' to the matcher i.e.: Match just if the property is from the child-class or other way around.